.github/workflows/kernelctf-auto-releaser.yaml

```diff
@@ -2,34 +2,158 @@ name: kernelCTF auto releaser
         uses: actions/checkout@v4
-
+      
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 'stable'
+      
       - name: Install prerequisites
-        run: sudo apt install -yq --no-install-recommends python3-lxml
-
-      - id: check
-        name: Check latest kernel versions
+        run: sudo apt-get install -yq --no-install-recommends jq python3-lxml
+      
+      - name: Build fetch_artifact tool
+        run: |
+          git clone https://android.googlesource.com/tools/fetch_artifact
+          cd fetch_artifact
+          go build -o fetch_artifact .
+          sudo mv fetch_artifact /usr/local/bin/
+          cd ..
+          rm -rf fetch_artifact
+      
+      - name: Restore last build ID cache
+        uses: actions/cache/restore@v3
+        with:
+          path: kernelctf/last_build_id.txt
+          key: android-build-id
+      
+      - name: Check latest Linux kernel versions
+        id: check_linux
         run: ./get_latest_kernel_versions.py
+      
+      - name: Check for new Android builds
+        id: check_android
+        run: |
+          API_URL="https://androidbuildinternal.googleapis.com/android/internal/build/v3/builds"
+          TARGET="aosp_cf_x86_64_only_phone-userdebug"
+          BRANCH="aosp-android-latest-release"
+          
+          # Fetch latest build (anonymous GET request)
+          echo "Fetching latest build from Android Build API..."
+          RESPONSE=$(curl -s "${API_URL}?branches=${BRANCH}&buildAttemptStatus=complete&buildType=submitted&maxResults=1&successful=true&target=${TARGET}")
+          
+          # Parse buildId
+          BUILD_ID=$(echo "$RESPONSE" | jq -r '.builds[0].buildId // empty')
+          
+          if [[ -z "$BUILD_ID" ]]; then
+            echo "Error: No builds found or failed to parse buildId"
+            echo "releases=[]" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+          
+          echo "Latest build ID: $BUILD_ID"
+          
+          # Check if build already processed
+          LAST_BUILD_ID=""
+          if [[ -f "last_build_id.txt" ]]; then
+            LAST_BUILD_ID=$(cat last_build_id.txt)
+            echo "Last processed build ID: $LAST_BUILD_ID"
+          fi
+          
+          if [[ "$LAST_BUILD_ID" == "$BUILD_ID" ]]; then
+            echo "No new builds available"
+            echo "releases=[]" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+          
+          echo "New build detected: $BUILD_ID"
+          
+          # Download kernel_version.txt using fetch_artifact
+          echo "Fetching kernel_version.txt..."
+          fetch_artifact \
+            -target="$TARGET" \
+            -build_id="$BUILD_ID" \
+            -artifact=kernel_version.txt
+          
+          if [[ ! -f "kernel_version.txt" ]]; then
+            echo "Error: kernel_version.txt not found"
+            echo "releases=[]" >> $GITHUB_OUTPUT
+            exit 0
```

Comment:
Fail the process with "exit 1", so we will a notification of the failed process and we can fix it.